packages/ui/Marquee/MarqueeDecorator.js

```diff
@@ -567,7 +567,9 @@ const MarqueeDecorator = hoc(defaultConfig, (config, Wrapped) => {
 				});
 			}
 
-			this.demote();
+			if (this.state.promoted) {
```

Comment:
Is it safe to skip all of `demote()` since it also stops the `promoteJob` or should this check just be applied around the `idle()` call?

Reply:
Doesn't seem like any change is required. I tried to repro and wasn't able to and couldn't find a way through reading the code that seemed to indicate a problem.

Reply:
I'm not so sure. I think I would prefer this to check the state in the `demote` method so we can safely abort the promotion job even if it hasn't finished yet.